serialization/serializer.go

```diff
@@ -161,7 +162,9 @@ func (s *serializer) flushToDisk(ctx actor.Context) error {
 	}()
 
 	// This maps strings to index position in a slice. This is doing to reduce the file size of the data.
-	strMapToIndex := make(map[string]uint32)
+	// Assume roughly each series has 10 labels, we do this because at very large mappings growing the map took up to 5% of cpu time.
+	// By pre allocating it that disappeared.
+	strMapToIndex := make(map[string]uint32, len(s.series)*10)
```

Comment:
Interesting. Kinda surprised it had so big impact. Maybe we can track the average size of this and autotune it at runtime?

Reply:
I was also surprised. This trades memory for cpu, ie this map is bigger than we actually need in most cases since we store only the unique strings.